### PR TITLE
RE-1150 Remove old RPC-O specific code

### PR DESCRIPTION
**Pull request by odyssey4me**
This removes jobs that have either been replaced or have not
been working for some time and their related code. It also
removes a bunch of RPC-O legacy code which is currently
unused.

Issue: [RE-1150](https://rpc-openstack.atlassian.net/browse/RE-1150)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_xenial_functional_test

**Comment by odyssey4me** [comment]
recheck_cit_re_unit_tests

**Comment by odyssey4me** [comment]
recheck_re_unit_tests

**Comment by odyssey4me** [comment]
recheck_unit_tests

**Comment by mattt416** [comment]
recheck_cit_re_unit_tests

**Comment by mattt416** [comment]
recheck_re_unit_tests

**Comment by mattt416** [comment]
recheck_unit_tests

**Comment by mattt416** [comment]
Wasn't the suggestion in https://github.com/rcbops/rpc-gating/pull/561 to break this into smaller PRs?

**Comment by odyssey4me** [comment]
@mattt416 Not smaller PR's, but smaller commits instead.

File: pipeline_steps/maas.groovy [hunk 1]
```diff
@@ -1,126 +0,0 @@
...
-    return sh (
-      script: """#!/bin/bash
-        cd ${env.WORKSPACE}/rpc-gating/scripts
-        set +x; . ${env.WORKSPACE}/.venv/bin/activate; set -x
-        ./maasutils.py --username ${env.PUBCLOUD_USERNAME} --api-key ${env.PUBCLOUD_API_KEY} ${args.join(" ")}
-      """,
-      returnStdout: true
-    )
-  }
-}
```
Comment (mattt416) [inline pipeline_steps/maas.groovy]: @git-harry @odyssey4me any likelihood that we'll use `get_maas_token_and_url` and `maas_utils` when deploying MaaS on "infra" nodes?

Comment (odyssey4me) [reply pipeline_steps/maas.groovy]: I expect we can always add it back if we need it.

**Pull request merged**